### ISSUE TITLE
(Feature request)  Please bring back the black theme

### ISSUE THREAD

**Issue by gerroon** [TriliumNext/Trilium#2273]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/zadam/trilium/issues) for a feature request that matches the one I want to file, without success.

### Describe feature

Hi

0.4.83 is missing the black theme that the previous versions had.  The black them was one of the reasons I was using Trilium, I am sad to see that this theme is not provided in the current upgrade. The current dark theme is very weak, the fonts are very hard to read.

I am going to downgrade personally because the current dark them is impossible to read for me, especially on Windows, the font antialiasing of the bold letters is insanely visible. The note tree font selection is nothing close to the original,  a lot of wasted space. I am sorry that I am being negative here but this upgrade is visually unbearable for  a user like me.

thanks

**Comment by gerroon** [comment]
Would it be possible to at least provide a way to change the background color override which defaults to that mid-gray in the gray theme?

**Comment by LorenAmelang** [comment]
Just want to add a request for the old Black theme! I agree strongly with gerroon, I was shocked to see the paleness of the new "dark". Yes, there are a lot of people with good eyes who like the softness, but there are a few of us with vision problems who need all the contrast we can get! 

Also, the "Zoom factor" seems a radical change...  I set it down to 0.6 to get about the same info on the screen as before. But then I have to make all the fonts way bigger, in all the different places that control them. Maybe it interacts with Windows screen scaling? I have 3000x2000 hardware, but scaled to "200%" so 1500x1000 for apps. 

Off to study the css file, instead of real work...  I could change it, but then what? Rebuild my own app? I don't see any separate css files in the install...  Is there some way to just override the existing css?

**Comment by zhuyizheng** [comment]
I agree with @LorenAmelang . I use the dark theme, but the dark theme seems to have been changed also. The contrast is lower than before. I would like to change the background color and the font color.

**Comment by LorenAmelang** [comment]
I've spent way too much of the last two days trying to work around the changes in the latest Trilium, Some things I've learned:

```
The "Zoom factor" seems a radical change...  Maybe it interacts with Windows screen scaling? 
I have 3000x2000 hardware, but scaled to "200%" so 1500x1000 for apps.
```

Following this old page, which works for most of the apps I've tried it with: 

https://danantonielli.com/adobe-app-scaling-on-high-dpi-displays-fix/

I installed a "manifest" to tell Windows not to scale Trilium, to show it in the 3000x2000 hardware space. Nothing changed. 

I noticed his update about new Windows controls for scaling:

https://danantonielli.com/app-scaling-on-high-dpi-displays-fix-2019/

That also did not change the overall size and layout at all. If you 

```
Check ‘Override high DPI scaling behavior‘ at the bottom
Select ‘System (Enhanced)‘ (or 'System') from the drop down
```

all the fonts and lines get slightly blurrier, but there are no size changes. 

Since Trilium now has internal settings for Zoom factor and font sizes, it would be great to be able to start from the maximum screen resolution. That works well in other apps I've tried. 


```
I'd asked: 
Is there some way to just override the existing css?
```

After much searching, I found 

https://github.com/zadam/trilium/wiki/Themes#custom-css

```
To do this, just create a code note with CSS type, put your custom CSS code into the note's content and create "appCss" label. 
When Trilium frontend starts, all notes with "appCss" label are appended in the style element of the Trilium HTML page.
```
I created some, tried all kinds of different statements to modify the font and background colors, and nothing made any difference. Copied bits of the built-in "Dark" theme and modified color values - refreshed, restarted, nothing changed. 


So I tried some of the themes listed at the bottom of the page:

```
https://github.com/Abourass/TriliumThemes
https://github.com/ZMonk91/Material-Dark-Trilium
https://github.com/raphwriter/trilium-theme-melon
https://github.com/SADAVA/trilium-notes-theme-dark-plus
```

Installed them, they show in the Menu -> Options -> Theme dropdown, but they all look exactly like the new Light theme! 

Maybe they are obsolete? The Themes page seems to be fairly recent - "zadam edited this page 23 days ago". 

```
v0.48.0-beta release Pre-release
@zadam zadam released this 26 days ago
```

Maybe the themes did not get checked for compatibility? 


So I made my own, and copied in the code from:

https://github.com/zadam/trilium/blob/master/src/public/stylesheets/theme-dark.css

"Latest commit 5938e03 26 days ago" so it should be the right one. But it looks exactly like the new Light theme! Hacking the values makes no difference. 


There must be a problem with the custom theme strategy...  

```
Custom CSS themes
Trilium provides a concept of custom user theme. You can make yourself one by creating a CSS code note and annotating it with #appTheme=my-theme-name label.
```

None of the https://github.com/ themes mention anything beyond "creating a CSS code note" and annotating it with "#appTheme". No mention of a "name" label, though maybe the name of the note serves that purpose - it is what shows in the Theme dropdown. 


```
It says: 
You can see an example of what you can put there below:
```
body.theme-my-theme {
    --main-background-color: #404552;
    --main-text-color: #AFB8C6;
    
I see, in note "melon-theme-3" with the Attribute "#appTheme":
body.theme-melon-theme-3

So if we are supposed to use:
body.theme-my-theme

Does that mean we must make the Note name "my-theme"? 

And is there some magic in adding the name to the Attribute: 
#appTheme=my-theme-name

And if so, where does the "-name" ending come from?

Tried exactly that, with the new Dark code surrounded by:

body.theme-official-dark {
}

No change from the Light theme. 


I noticed in 

https://github.com/Engr-AllanG/trilium-themes#notes

```
You need to add an attribute of type Label with name appTheme and no value. After setting the attribute, you need to select the theme in options. You may need to re-load the front end (ctrl-r).

Issues
Switching to another theme does not work entirely without removing the appTheme attribute. I don't know why.
```

Does that suggest you can only have one user theme at a time? Or that they interact somehow? 


The Log seems no help:

```
14:20:01.117 200 GET /api/options/user-themes with 354 bytes took 4ms
14:20:08.416 Updating option theme to official-dark-name
14:20:08.417 204 PUT /api/options/theme/official-dark-name with 0 bytes took 2ms
14:20:08.432 JS Info: Frontend app reload: theme change
14:20:08.445 Generated CSRF token dLn6H2oN-Y7WZHsMF2fKpfDMJ4Ak-hAVQKSQ with secret undefined
```

(Still only shows "official-dark" in the dropdown.) 

And logs the same lines of info when I choose the internal Dark theme: 

```
14:20:26.499 200 GET /api/options/user-themes with 354 bytes took 6ms
14:20:33.403 Updating option theme to dark
14:20:33.404 204 PUT /api/options/theme/dark with 0 bytes took 1ms
14:20:33.420 JS Info: Frontend app reload: theme change
14:20:33.433 Generated CSRF token ZFTGvjmB-KlodvONoLqhAX3fLBHje3ViBUQQ with secret undefined
```


It does keep showing this - is it a problem?

```
14:23:22.050 200 GET /api/tree with 6076 bytes took 0ms
14:23:22.051 Keyboard action showNoteInfo found in database, but not in action definition.
14:23:22.051 Keyboard action showLinkMap found in database, but not in action definition.
14:23:22.051 Keyboard action toggleZenMode found in database, but not in action definition.
14:23:22.051 Keyboard action focusOnAttributes found in database, but not in action definition.
14:23:22.052 200 GET /api/keyboard-actions with 11944 bytes took 1ms

```

Background info:

```
13:04:17.072 {
  "appVersion": "0.48.5",
  "dbVersion": 185,
  "syncVersion": 21,
  "buildDate": "2021-11-01T10:25:43+01:00",
  "buildRevision": "6f60cf1a861fb9b5a4af5f4f916b6d0e416bef64",
  "dataDirectory": "C:\\Users\\loren\\AppData\\Roaming\\trilium-data",
  "clipperProtocolVersion": "1.0"
}
13:04:17.073 CPU model: Intel(R) Core(TM) i7-6600U CPU @ 2.60GHz, logical cores: 4 freq: 2808 Mhz
```


I'd really appreciate any clues toward going back to high contrast dark mode!

**Comment by crameth** [comment]
Yes, it's fairly easy to theme Trilium as long as you have a bit of CSS knowledge. The setup process for installing/creating your own theme is still the same as before:

1. Create an empty note anywhere in Trilium
2. Copy and paste the CSS styles found here: https://github.com/crameth/trilium-theme-another-black/blob/main/theme-another-black.css
3. Set the note type to "CSS" under basic properties (1st button from the left).
![image](https://user-images.githubusercontent.com/2794843/140922849-07307f1f-a871-4f7a-88e2-f8687e7a2357.png)
4. Under owned attributes (2nd button from the left), assign the value `#appTheme` with the theme name you like (in this case I used "Black").
![image](https://user-images.githubusercontent.com/2794843/140922678-e13ecb6f-d86b-4193-a4c9-b6e784a50d98.png)

Reload the frontend just to be safe, then select the new theme.

**Comment by gerroon** [comment]
Thanks for the tutorial I was able to apply it, but the fonts in some notes looks like shit :( Can anyone even read  this easily?

![Uploading trilium_BlDGBlHZk0.jpg…]()

**Comment by crameth** [comment]
The fonts pictured are only for code-type notes and it's using the default mono space font. You can try applying your own mono space font if it's not readable.

**Comment by surajsharma** [comment]
> I don't plan to bring it back.

shame, i preferred the old one.. it was retro, kindly reconsider?

**Comment by gerroon** [comment]
> The fonts pictured are only for code-type notes and it's using the default mono space font. You can try applying your own mono space font if it's not readable.

How would I do that with this https://github.com/crameth/trilium-theme-another-black/blob/main/theme-another-black.css ?

**Comment by cwilliams5** [comment]
Hey all. I wasn't happy with the built in dark theme or any community ones, which had any number of problems (incomplete, fuzzy fonts due to filters, bad contrast, etc, etc). So I made my own, Midnight. Turned out nice enough I figured I'd share it since I got my theming start in this thread. 

> I don't plan to bring it back.
> 
If it's a matter of maintenance and you're at all interested, I'd be happy to PR Midnight and maintain it. 

## [Midnight-Trilium-Dark-Mode](https://github.com/cwilliams5/Midnight-Trilium-Dark-Mode)

### Features
* Material Design Based
  * Comfortable Contrast
  * Maximum Readability
* Dark Mode Friendly Font
* Consistent Design
* Custom Code View Styling
  * Dark Mode Native
  * Ligature Font
* Complete Style Coverage
  * More than built in “dark” mode!

### Screenshot
![Midnight Screenshot](https://raw.githubusercontent.com/cwilliams5/Midnight-Trilium-Dark-Mode/main/midnight.png "Midnight Screenshot")

**Comment by eliandoran** [comment]
We have a new theme however we don't plan to support a black theme officially, it needs to be handled as a custom theme.
